### PR TITLE
Justin steven rbtree deletion

### PR DESCRIPTION
**Pull request by StevenJiao**
rb tree insertions and deletions complete

### CODE REVIEW

**Comment by jboileau99** [comment]
LETS GOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

File: RBTree/src/main.rs [hunk 1]
```diff
@@ -1,10 +1,13 @@
-use std::cell::RefCell;
```
Comment (christianjans) [inline RBTree/src/main.rs:1]: Looks awesome! Can we get Matt's command-line stuff back in main? Or are you still using main for something?

Comment (StevenJiao) [reply RBTree/src/main.rs:1]: that's my bad I forgot to clean it up. Main should be good to go now!

**Comment by StevenJiao** [comment]
also accounts for #3 for RBtrees

**Review by christianjans — APPROVED** [review]

**Comment by StevenJiao** [comment]
@christianjans I just added a deletion case for single root and a test case for it, lemme know if theres any issues and I'll push it into main after its approved

**Comment by christianjans** [comment]
@StevenJiao, the tests all pass on my machine for the RB tree.. is that what you?

**Comment by StevenJiao** [comment]
@christianjans sorry I meant in lines 516-524 changes in lib.rs, I added in the logic so that the tree would delete a root with a single node (no parent, no children) as it wasn't in there before. There is also the added test case for testing it in lines 800-809 for testing it. let me know if these new changes mess anything up for you and I'll merge it after

**Comment by christianjans** [comment]
@StevenJiao, oh no that should be fine! I had to do something similar in AVL delete.

**Pull request merged**